diff --git a/interface/vsomeip/constants.hpp b/interface/vsomeip/constants.hpp
index 2b040c5..112d155 100644
--- a/interface/vsomeip/constants.hpp
+++ b/interface/vsomeip/constants.hpp
@@ -13,7 +13,8 @@

 namespace vsomeip_v3 {

-const major_version_t DEFAULT_MAJOR = 0x00;
+//const major_version_t DEFAULT_MAJOR = 0x00;
+const major_version_t DEFAULT_MAJOR = 0x01; // needed for AUTOSAR compatibility
 const minor_version_t DEFAULT_MINOR = 0x00000000;
 const ttl_t DEFAULT_TTL = 0xFFFFFF; // "until next reboot"

